sympy/solvers/solvers.py

```diff
@@ -863,64 +903,100 @@ def _sympified_list(w):
                         symbols.add(p)
                         pot.skip()  # don't go any deeper
         symbols = list(symbols)
-
         ordered_symbols = False
-    elif len(symbols) == 1 and iterable(symbols[0]):
-        symbols = symbols[0]
+    else:
+        # User has given symbols as *args or as an iterable arg
+        if len(symbols) == 1 and iterable(symbols[0]):
+            symbols = symbols[0]
+            if isinstance(symbols, GeneratorType):
+                symbols = list(symbols)
+        # Has the user given an unordered collection like a set?
+        ordered_symbols = is_sequence(symbols)
+        symbols = [sympify(sym) for sym in symbols]
+
+    # Remove symbols the user is not interested in. This removes symbols even
+    # if they were explicitly specified.
 
-    # remove symbols the user is not interested in
-    exclude = flags.pop('exclude', set())
     if exclude:
-        if isinstance(exclude, Expr):
+        if not iterable(exclude):
             exclude = [exclude]
         exclude = set().union(*[e.free_symbols for e in sympify(exclude)])
-    symbols = [s for s in symbols if s not in exclude]
+        symbols = [s for s in symbols if s not in exclude]
 
+    # alternate exits
 
-    # preprocess equation(s)
-    ###########################################################################
-    for i, fi in enumerate(f):
-        if isinstance(fi, (Equality, Unequality)):
-            if 'ImmutableDenseMatrix' in [type(a).__name__ for a in fi.args]:
-                fi = fi.lhs - fi.rhs
-            else:
-                L, R = fi.args
-                if isinstance(R, BooleanAtom):
-                    L, R = R, L
-                if isinstance(L, BooleanAtom):
-                    if isinstance(fi, Unequality):
-                        L = ~L
-                    if R.is_Relational:
-                        fi = ~R if L is S.false else R
-                    elif R.is_Symbol:
-                        return L
-                    elif R.is_Boolean and (~R).is_Symbol:
-                        return ~L
-                    else:
-                        raise NotImplementedError(filldedent('''
-                            Unanticipated argument of Eq when other arg
-                            is True or False.
-                        '''))
+    if any(_is_eq_bool(fi) for fi in f):
+        for i, fi in enumerate(f):
+            if not isinstance(fi, (Equality, Unequality)):
+                continue
```

Comment:
What exactly does this block do? It looks similar to `_invert_eq_bool` which is used below.

Reply:
The two lines 
```
if not isinstance(fi, (Equality, Unequality)):
    continue
```
doesn't seem to do anything and can be dropped.

Reply:
I mean the whole block inside the `if any(_is_eq_bool(fi) for fi in f):`

Reply:
> What exactly does this block do? 

This handles cases like `solve(Eq(x, True))`

Reply:
The purpose of the block is pretty much the same like `_invert_eq_bool()`. Except that the block has a few additional checks. We could modify `_invert_eq_bool()` to replace the block.

Reply:
I think this block is pretty much pointless but I don't suggest to remove it right now. If you can factor it out entirely into a separate function then that would be good.

Reply:
I can't. Factoring the block out breaks on thing or another. I guess this should better be deleted, in another PR maybe.